pkg/resources/globalrolebinding/globarolebinding.go

```diff
@@ -36,6 +40,27 @@ func (grbv *globalRoleBindingValidator) Admit(response *webhook.Response, reques
 	// Pull the global role to get the rules
 	globalRole, err := grbv.globalRoles.Get(newGRB.GlobalRoleName)
 	if err != nil {
+		if errors.IsNotFound(err) {
+			switch request.Operation {
+			// allow delete operations if the GR associated to the GRB is not found
+			case admissionv1.Delete:
+				response.Allowed = true
+				return nil
+			// only allow updates to the finalizers if the GR associated to the GRB is not found
+			case admissionv1.Update:
+				if newGRB.DeletionTimestamp != nil {
+					response.Allowed = true
+					return nil
+				}
+				response.Result = &metav1.Status{
+					Status:  "Failure",
+					Message: fmt.Sprintf("GlobalRoleBinding %s associated GlobalRole not found. Only deletion and updates to finalizers are permitted.", newGRB.Name),
```

Comment:
Errors should start with lowercase and this message is confusing. Can it be updated to something like `referenced globalRole %v not found, updates not allowed`. I don't think we need to call out the finalizer stuff as that's an implementation detail of rancher and how we object lifecycles and probably won't be useful to a user.

Reply:
Yes, agreed, I was trying to figure out a good way to say this. I'll change it to this